R/process_genomic_data.R

```diff
@@ -0,0 +1,134 @@
+#' Function to perform the genome assembly
+#' @param input.dir the path to the folder with the input fastq files. Each sample/barcode will have a directory in this folder
+#' @param assembly.dir the path to the folder where the output files from the genome assembly will be stored
+#' @param consensus.fasta.dir the path to the folder where the merged consensus fasta file will be stored
+#' @param num.samples the number of samples in the dataset
+#' @param device the sequencing device (either MinION or PromethION)
+#' @param partition the partition. Only set this when running on a HPC, default is NULL
+#' @returns the output of this function is the merged fasta file that will be in the folder specified using the consensus.fasta.dir parameter
+#' @export
+# @importFrom foreach dopar
+genome_assembly <- function(input.dir, assembly.dir, consensus.fasta.dir, num.samples, device = "MinION", partition = NULL) {
+  # @examples genome_assembly(input.dir="/Users/karimmane/Documents/Karim/LSHTM/TRACE_dev/Genomic_data/Gam_10", assembly.dir="/Users/karimmane/Documents/Karim/LSHTM/TRACE_dev/Genomic_data/Gam10_assembly", consensus.fasta.dir="/Users/karimmane/Documents/Karim/LSHTM/TRACE_dev/Genomic_data/Gam10_consensus", num.samples=5, device="MinION", partition=NULL)
+  # checking the input parameters
+  if (!dir.exists(input.dir)) {
+    stop(input.dir, " not found!")
+  }
+  system(sprintf("mkdir -p %s %s", assembly.dir, consensus.fasta.dir))
+  if (!is.numeric(num.samples)) {
+    stop("The value for the num.samples argument should be type numeric.")
+  }
+  if (!(device %in% c("MinION", "PromethION"))) {
+    stop("Incorrect sequencing device name provided! Expected value are: 'MinION' or 'PromethION'")
+  }
+  if (device == "MinION") {
+    medaka.model <- "r941_min_fast_g303"
+  } else {
+    medaka.model <- "r941_prom_hac_g303"
+  }
```

Comment:
Suggest using `switch()` here, which could make it easier to support other options in the future.